Design Patterns/Overview/Document.md

```diff
@@ -0,0 +1,60 @@
+# Design Pattern
+
+Design patterns are general solutions to commonly occurring problems in a context (software development)
+
+## Why should we learn design patterns?
+
+1. Avoid re-inventing wheel: There is already an existing solution to the problem that arises in software development so using that saves time.
+
+1. Better Communication: It is easier to communicate via design pattern name our intent over a code rather than through low-level discussion
+
+ Example : 
+- Conversation one: As the database access layer is tightly coupled to the database we can extract an interface and put the necessary method declaration in it. Implementing the interface in the class and providing a definition help us to replace all the local instantiation of the object with an interface 
+- Conversation Two : As the database access layer is tightly coupled to the database we can go for a repository pattern
+- Conversation two feels good and better than conversation one
+
+1. Better and optimal code: Help us write a clean and more optimal code that boosts the performance of the application
+
+## Document structure for every pattern
+
+- Name: name of the pattern
+- Intent: Intent or goal it trying to achieve
+- Alias: Other name for the pattern if exists
+- Motivation/applicability or scenario: The context or scenerio the pattern suits very well.
+- Participant : what are the classes or interface etc., it going to have
+- Collabration : How the participant going to interface with each other
```

Comment:
not able to understand

Reply:
Please be more specific

Reply:
Collabration : How the participant going to interface with each other  *interface

Reply:
>  Collaboration : How the participants are going to interface with each other

Collaboration : How the participants are going to interact with each other

I will correct it and update